JASP-Engine/JASPgraphs/R/printJASPgraphs.R

```diff
@@ -0,0 +1,41 @@
+# print.JASPgraphs <- function(x, newpage = is.null(vp), vp = NULL, ...) {
+#
+#   tmp <- mget(x = c("x_custom", "y_custom", "element_grob.element_custom_x", "element_grob.element_custom_y"),
+#               envir = .GlobalEnv, ifnotfound = rep(list(NULL), 4))
+#
+#   assign("x_custom", JASPgraphs:::x_custom, envir = .GlobalEnv)
+#   assign("y_custom", JASPgraphs:::y_custom, envir = .GlobalEnv)
+#   assign("element_grob.element_custom_x", JASPgraphs:::element_grob.element_custom_x, envir = .GlobalEnv)
+#   assign("element_grob.element_custom_y", JASPgraphs:::element_grob.element_custom_y, envir = .GlobalEnv)
+#
+#   x <- ggplot2:::print.ggplot(x, newpage = newpage, vp = vp, ...)
+#
+#   assign("x_custom", tmp[[1L]], envir = .GlobalEnv)
+#   assign("y_custom", tmp[[2L]], envir = .GlobalEnv)
+#   assign("element_grob.element_custom_x", tmp[[3L]], envir = .GlobalEnv)
+#   assign("element_grob.element_custom_y", tmp[[4L]], envir = .GlobalEnv)
+#
+#   invisible(x)
+#
+# }
+
+#' @method print JASPgraphs
+#' @export
+print.JASPgraphs <- function(x, ...) {
+
+  if (ggplot2::is.ggplot(x)) {
+    ggplot2:::print.ggplot(x, ...)
+  } else if (inherits(x, c("gtable", "gTree", "grob", "gDesc"))) {
+    gridExtra::grid.arrange(x, ...)
+  } else {
+    stop(sprintf(
```

Comment:
Does R support a way to call the next S3 method? Because this doesn't fail very gracefully, eventhough a different class might have an appropriate print method defined.

Reply:
what do you mean? We could do something like

```
class(x) <- class(x)[-1]
e <- try(
	print(x) # hope for the best
)
if (inherits(e, "try-error")) {
  stop("...")
} else {
  return(invisible(e))
}
```
instead? 

This function is only called whenever people use `print(jaspGraphsObject)` we (or I) better make sure it works for objects we've created.

Reply:
the function `NextMethod()`, immediately calls the next S3 method btw.

Reply:
I would say use `NextMethod()` given that `length(class(x)) > 1`. The error seems a bit overkill. Mainly thinking about future cases where 'random' people might work with our objects and  wish to extend them.
Up to you though, it's not that big of a deal.